File: library/src/main/java/renetik/android/CSLocale.java
package renetik.android;

import java.util.Locale;

public class CSLocale {

	public static boolean isMetric() {
		return Locale.getDefault() != Locale.US;
	}

	public static double toCelsius(int fahrenheit) {
		return (fahrenheit - 32) / 1.8;
	}

	public static double toFahrenheit(int celsius) {
		return celsius * 1.8 + 32;
	}

	public static double fromCelsiusToLocale(int celsius) {
		return isMetric() ? celsius : toFahrenheit(celsius);
	}

	public static double fromFahrenheitToLocale(int fahrenheit) {
		return !isMetric() ? fahrenheit : toCelsius(fahrenheit);
	}

}

File: library/renetik-android-core/src/main/java/renetik/android/core/util/CSLocale.kt
package renetik.android.core.util

import java.util.*

object CSLocale {

    val isMetric: Boolean
        get() = Locale.getDefault() !== Locale.US

    fun toCelsius(fahrenheit: Int): Double {
        return (fahrenheit - 32) / 1.8
    }

    fun toFahrenheit(celsius: Int): Double {
        return celsius * 1.8 + 32
    }

    fun fromCelsiusToLocale(celsius: Int): Double {
        return if (isMetric) celsius.toDouble() else toFahrenheit(celsius)
    }

    fun fromFahrenheitToLocale(fahrenheit: Int): Double {
        return if (!isMetric) fahrenheit.toDouble() else toCelsius(fahrenheit)
    }

}
